### PR TITLE
Add GenerateFormItem component for automatic generation basic FormItem

### PR DESCRIPTION
**Pull request by dessli**
### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [x] Demo update
- [ ] Component style update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 💡 Background and solution

Simplify the construction of basic forms or multi-field forms.

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

File: components/form/GenerateFormItem.razor.cs [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+    [CascadingParameter(Name = "Form")]
+    private IForm Form { get; set; }
+
+
+    /// <summary>
+    /// If true then the rules will be required by CheckTypeRequire.
+    /// Recommended only for forms without complex validation, where only simple validation for empty values is required.
+    /// </summary>
+    [Parameter]
+    public bool RulesRequireByType { get; set; }
```
Comment (ElderJames) [inline components/form/GenerateFormItem.razor.cs:31]: I think we don't need this parameter and use `ValidateMode` instead to set the validation.

Comment (dessli) [reply components/form/GenerateFormItem.razor.cs:31]: I agree with your perspective. We can remove this parameter and use ValidateMode instead to set the validation. I’ll make the necessary adjustments to the code, or have you already started making the changes?

Comment (ElderJames) [reply components/form/GenerateFormItem.razor.cs:31]: Not yet, please.

File: components/form/GenerateFormItem.razor.cs [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+        if (underlyingType == typeof(DateTime))
+        {
+            return MakeDatePicker(property, isRequired);
+        }
+
+        if (underlyingType.IsEnum)
+        {
+            return MakeEnumSelect(property);
+        }
+
```
Comment (ElderJames) [inline components/form/GenerateFormItem.razor.cs:118]: I think we also need to consider nested objects.

File: components/form/GenerateFormItem.razor.cs [hunk 3]
```diff
@@ -22,13 +22,8 @@ public partial class GenerateFormItem<TModel> : ComponentBase
...
     private IForm Form { get; set; }
 
-
-    /// <summary>
-    /// If true then the rules will be required by CheckTypeRequire.
-    /// Recommended only for forms without complex validation, where only simple validation for empty values is required.
-    /// </summary>
     [Parameter]
-    public bool RulesRequireByType { get; set; }
+    public Func<PropertyInfo, FormValidationRule[]?> ValidateRules { get; set; }
```
Comment (dessli) [inline components/form/GenerateFormItem.razor.cs:26]: @ElderJames I have removed RulesRequireByType and added ValidateRules.

File: components/form/GenerateFormItem.razor.cs (hunk 2 above)
Comment (dessli) [reply components/form/GenerateFormItem.razor.cs:107]: I’ll think about how to handle this.

File: components/form/GenerateFormItem.razor.cs [hunk 4]
```diff
@@ -52,23 +47,29 @@ protected override void BuildRenderTree(RenderTreeBuilder builder)
             }
 
             var displayName = property.GetCustomAttribute<DisplayNameAttribute>()?.DisplayName ?? property.Name;
```
Comment (ElderJames) [inline components/form/GenerateFormItem.razor.cs:49]: We can use `PropertyReflector` to get the display name from localization.

File: components/form/GenerateFormItem.razor.cs [hunk 5]
```diff
@@ -52,23 +47,29 @@ protected override void BuildRenderTree(RenderTreeBuilder builder)
...
-            childContent ??= GenerateByType(underlyingType, property, displayName, isRequired);
+            childContent ??= GenerateByType(underlyingType, property, displayName);
 
             // Make FormItem 
-            if (isRequired)
+            if (validateRule != null)
             {
-                var formValidation = GetFormValidationRuleByType(underlyingType);
                 builder.OpenComponent(0, typeof(FormItem));
                 builder.AddAttribute(1, "Label", displayName);
```
Comment (ElderJames) [inline components/form/GenerateFormItem.razor.cs:70]: The Label would auto generated inside FormItem.

**Comment by dessli** [comment]
@ElderJames  Hi, I’ve refactored the code to support the automatic generation of nested types. For the automatic generation of array classes, I’m considering using Dynamic Field in the future.

**Comment by ElderJames** [comment]
That's very cool @dessli ! And yes, using Dynamic Fields can reduce some of the code, and users can customize the components they want.

**Comment by ElderJames** [comment]
And I have an idea that we can use [Collapse](https://antblazor.com/en-US/components/collapse) to wrap the sub form generated by nested types. So that the style will be better.

**Comment by dessli** [comment]
> And I have an idea that we can use [Collapse](https://antblazor.com/en-US/components/collapse) to wrap the sub form generated by nested types. So that the style will be better.

ok, I’m considering a feature for multi-style selection as well.

**Comment by dessli** [comment]
@ElderJames Hi, I’ve added the multi-style feature, which currently supports Collapse and Block styles. The default style is set to Collapse.

**Comment by ElderJames** [comment]
Looks like validation is not working in nest block.
image

**Comment by dessli** [comment]
> Looks like validation is not working in nest block. image

It seems to be caused by the subclass reflection not being properly bound to the component. I’ll make corrections later. I’ve been really busy with work lately.

**Comment by dessli** [comment]
@ElderJames Hi, The validation is not working problem, I’ve found the cause. It was an issue with the demo code where I forgot to add ValidateMode and ValidateComplexType.

**Comment by ElderJames** [comment]
All right. But I think the required attribute is by default.

**Review by ElderJames — APPROVED** [review]

**Comment by ElderJames** [comment]
Thank you for contribution @dessli .

**Pull request merged**